src/index.js

```diff
@@ -708,24 +716,44 @@ class LocalstackPlugin {
    *                    a rejected promise on any connection error.
    */
   checkTCPConnection(options) {
-    return new Promise((resolve, reject) => {
-      const socket = new net.Socket();
-      const client = socket.connect(options, () => {
-        client.end();
-        resolve();
-      });
+    let waitForConnectionTask = () =>
+      new Promise((resolve, reject) => {
+        const socket = new net.Socket();
+        const client = socket.connect(options, () => {
+          client.end();
+          resolve();
+        });
 
-      client.setTimeout(500);  // milliseconds
-      client.on("timeout", err => {
-        client.destroy();
-        reject(err);
-      });
+        client.setTimeout(500);  // milliseconds
+        client.on("timeout", err => {
+          client.destroy();
+          reject(err);
+        });
 
-      client.on("error", err => {
-        client.destroy();
-        reject(err);
+        client.on("error", err => {
+          client.destroy();
+          reject(err);
+        });
       });
-    });
+
+    let waitForAsync = fn => {
+      let iterator = fn();
+      let loop = result => {
+        !result.done && result.value.then(
+          res => loop(iterator.next(res)),
+          err => loop(iterator.throw(err))
+        )
+      }
+      loop(iterator.next());
+    }
+
+    waitForAsync(function* () {
```

Comment:
Tested using the localstack-pro-sample [lambda-xray](https://github.com/localstack/localstack-pro-samples/tree/master/lambda-xray) (also works in community without X-Ray feature) with some test variables for the `hello` function:

```
    environment:
      TABLE_NAME: tableName
      SLS_AWS_ACCOUNT_ID: ${aws:accountId}
      SLS_AWS_REGION: ${aws:region}
```


✅ I can confirm that this `waitForAsync` busy-wait fixes the variable resolver issue when using the IPv6 workaround `host: http://127.0.0.1`
❎ Unfortunately, it re-introduces the IPv6 issue again but now during variable resolve time when trying to connect to localhost. The debug logs reveal that IPv6 is the issue `address: '::1'`:

```
sls deploy --stage local
Running "serverless" from node_modules
Using serverless-localstack
(node:86411) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
Recoverable error occurred (Inaccessible host: `localhost' at port `4566'. This service may not be available in the `us-west-2' region.), sleeping for ~4 seconds. Try 1 of 4
Recoverable error occurred (Inaccessible host: `localhost' at port `4566'. This service may not be available in the `us-west-2' region.), sleeping for ~5 seconds. Try 2 of 4
Recoverable error occurred (Inaccessible host: `localhost' at port `4566'. This service may not be available in the `us-west-2' region.), sleeping for ~5 seconds. Try 3 of 4
Recoverable error occurred (Inaccessible host: `localhost' at port `4566'. This service may not be available in the `us-west-2' region.), sleeping for ~5 seconds. Try 4 of 4
Environment: darwin, node 19.1.0, framework 3.28.1 (local) 3.27.0v (global), plugin 6.2.3, SDK 4.3.2
Credentials: Local, "default" profile
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Cannot resolve serverless.yml: Variables resolution errored with:
  - Cannot resolve variable at "functions.hello.environment.SLS_AWS_ACCOUNT_ID": Inaccessible host: `localhost' at port `4566'. This service may not be available in the `us-west-2' region.
```